parquetschema/autoschema/gen.go

```diff
@@ -0,0 +1,399 @@
+						IsSigned: false,
+					},
+				},
+			},
+		}, nil
+	case reflect.Uint16:
+		return &parquetschema.ColumnDefinition{
+			SchemaElement: &parquet.SchemaElement{
+				Type:           parquet.TypePtr(parquet.Type_INT32),
+				Name:           fieldName,
+				RepetitionType: parquet.FieldRepetitionTypePtr(parquet.FieldRepetitionType_REQUIRED),
+				ConvertedType:  parquet.ConvertedTypePtr(parquet.ConvertedType_UINT_16),
+				LogicalType: &parquet.LogicalType{
+					INTEGER: &parquet.IntType{
+						BitWidth: 16,
+						IsSigned: false,
+					},
+				},
+			},
+		}, nil
+	case reflect.Uint32:
+		return &parquetschema.ColumnDefinition{
+			SchemaElement: &parquet.SchemaElement{
+				Type:           parquet.TypePtr(parquet.Type_INT32),
+				Name:           fieldName,
+				RepetitionType: parquet.FieldRepetitionTypePtr(parquet.FieldRepetitionType_REQUIRED),
+				ConvertedType:  parquet.ConvertedTypePtr(parquet.ConvertedType_UINT_32),
+				LogicalType: &parquet.LogicalType{
+					INTEGER: &parquet.IntType{
+						BitWidth: 32,
+						IsSigned: false,
+					},
+				},
+			},
+		}, nil
+	case reflect.Uint64:
+		return &parquetschema.ColumnDefinition{
+			SchemaElement: &parquet.SchemaElement{
+				Type:           parquet.TypePtr(parquet.Type_INT64),
+				Name:           fieldName,
+				RepetitionType: parquet.FieldRepetitionTypePtr(parquet.FieldRepetitionType_REQUIRED),
+				ConvertedType:  parquet.ConvertedTypePtr(parquet.ConvertedType_UINT_64),
+				LogicalType: &parquet.LogicalType{
+					INTEGER: &parquet.IntType{
+						BitWidth: 64,
+						IsSigned: false,
+					},
+				},
+			},
+		}, nil
+	case reflect.Uintptr:
+		return nil, errors.New("unsupported type uintptr")
+	case reflect.Float32:
+		return &parquetschema.ColumnDefinition{
+			SchemaElement: &parquet.SchemaElement{
+				Type:           parquet.TypePtr(parquet.Type_FLOAT),
+				Name:           fieldName,
+				RepetitionType: parquet.FieldRepetitionTypePtr(parquet.FieldRepetitionType_REQUIRED),
+			},
+		}, nil
+	case reflect.Float64:
+		return &parquetschema.ColumnDefinition{
+			SchemaElement: &parquet.SchemaElement{
+				Type:           parquet.TypePtr(parquet.Type_DOUBLE),
+				Name:           fieldName,
+				RepetitionType: parquet.FieldRepetitionTypePtr(parquet.FieldRepetitionType_REQUIRED),
+			},
+		}, nil
+	case reflect.Complex64:
+		return nil, errors.New("unsupported type complex64")
+	case reflect.Complex128:
+		return nil, errors.New("unsupported type complex128")
+	case reflect.Array:
+		if fieldType.Elem().Kind() == reflect.Uint8 {
+			typeLen := int32(fieldType.Len())
+			// handle special case for [N]byte
+			return &parquetschema.ColumnDefinition{
+				SchemaElement: &parquet.SchemaElement{
+					Type:           parquet.TypePtr(parquet.Type_FIXED_LEN_BYTE_ARRAY),
+					Name:           fieldName,
+					RepetitionType: parquet.FieldRepetitionTypePtr(parquet.FieldRepetitionType_REQUIRED),
+					TypeLength:     &typeLen,
+				},
+			}, nil
+		}
+		// TODO: implement this
```

Comment:
WDYT of adding the missing types here? at least int/int16/int32/int64

Reply:
I looked at what we're doing in `floor`, and essentially, we're treating Go arrays the same as Go slices, so I just unified the code here and treat arrays as if they were slices, even though Go arrays don't translate particularly well to any parquet types.